spec/models/job_proxy_dispatcher_spec.rb

```diff
@@ -172,5 +175,103 @@ module JobProxyDispatcherSpec
         end
       end
     end
+
+    context "with container and vms jobs" do
+      before(:each) do
+        @jobs = (@vms + @repo_vms + @container_images).collect(&:scan)
+        @dispatcher = JobProxyDispatcher.new
+      end
+
+      describe "#pending_jobs" do
+        it "returns only vm jobs by default" do
+          jobs = @dispatcher.pending_jobs
+          expect(jobs.count).to eq(@vms.count + @repo_vms.count)
+          expect(jobs.all? { |x| x.target_class == 'VmOrTemplate' }).to eq(true)
```

Comment:
If this fails, does it say expected false to be true? 
If so, maybe we should do the expect for each job.